docs/notebooks/longrangecnot/LongRangeCNOTgates.ipynb

```diff
@@ -0,0 +1,2139 @@
+    "\n",
+    "import random\n",
+    "from IPython.display import clear_output, display\n",
+    "\n",
+    "import numpy as np\n",
+    "from numpy import pi\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "import pickle\n",
+    "\n",
+    "# Importing standard Qiskit libraries\n",
+    "from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister, transpile\n",
+    "from qiskit.visualization import *\n",
+    "\n",
+    "from qiskit_ibm_runtime import QiskitRuntimeService\n",
+    "\n",
+    "from qiskit.circuit import Gate\n",
+    "from qiskit.circuit.library import XGate\n",
+    "\n",
+    "from qiskit.providers.backend import BackendV2 as Backend\n",
+    "from qiskit.transpiler import CouplingMap, InstructionDurations\n",
+    "from qiskit.transpiler.passmanager import PassManager\n",
+    "from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager\n",
+    "from qiskit_ibm_provider.transpiler.passes.scheduling import DynamicCircuitInstructionDurations\n",
+    "from qiskit_ibm_provider.transpiler.passes.scheduling import ALAPScheduleAnalysis\n",
+    "from qiskit_ibm_provider.transpiler.passes.scheduling import PadDynamicalDecoupling\n",
+    " \n",
+    "from qiskit.circuit.classical import expr\n",
+    "\n",
+    "from qiskit.quantum_info import Pauli, PauliList\n",
+    "from qiskit.result import marginal_counts\n",
+    "from qiskit.visualization.timeline import draw\n",
+    "\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import warnings\n",
+    "warnings.filterwarnings(\"ignore\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22aae92a-2df7-4725-9eba-f41a783c2974",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "## Initialization for Account\n",
+    "\n",
+    "In order to run this notebook you will need an account on the [IBM Quantum Platform](https://quantum-computing.ibm.com/). More details on how to initialize your account can be found at [qiskit Runtime](https://www.ibm.com/quantum/qiskit-runtime)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7e5f25ed-a6f8-4bc4-b1f6-814278bc3a8e",
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "from qiskit_ibm_provider import IBMProvider\n",
+    "\n",
+    "# Save your credentials on disk.\n",
+    "# IBMProvider.save_account(token='<IBM Quantum API key>')\n",
+    "\n",
+    "provider = IBMProvider(instance='ibm-q-internal/deployed/default')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c75dbecb-c273-4955-991e-557f9b5a41df",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "## The Experiments\n",
+    "\n",
+    "In this notebook we will run a teleportation circuit in three different setups, where we always assume we have a line of n qubits (for varying n with n-2 empty ancillas in the middle and a CNOT that we’d like to apply between the two ends):\n",
```

Comment:
"teleportation circuit" do you mean something like "gate teleportation circuit"?

Reply:
Yes. Corrected